packages/client/hmi-client/src/workflow/ops/calibrate-julia/tera-calibrate-julia.vue

```diff
@@ -226,6 +285,177 @@ onMounted(() => {
+	const runIds: string[] = querySimulationInProgress(props.node);
+	if (runIds.length === 0) return;
+
+	if (runIds.includes(event.data.id)) {
+		const { iter, loss, params, solData, timesteps } = event.data;
+
+		parameterResult.value = filterStateVars(params);
+
+		lossValues.push({ iter, loss });
+		if (drilldownLossPlot.value) {
+			const width = drilldownLossPlot.value.offsetWidth;
+			renderLossGraph(drilldownLossPlot.value, lossValues, { width, height: 150 });
+		}
+
+		if (iter % 100 === 0) {
+			currentIntermediateVals.value = { timesteps, solData };
+		}
+	}
+};
+
+const getStatus = async (simulationIds: string[]) => {
+	showSpinner.value = true;
+	runInProgress.value = simulationIds[0];
+
+	await subscribeToUpdateMessages(
+		simulationIds,
+		ClientEventType.SimulationSciml,
+		getMessageHandler
+	);
+
+	poller
+		.setInterval(3000)
+		.setThreshold(300)
+		.setPollAction(async () => simulationPollAction(simulationIds, props.node, progress, emit));
+	const pollerResults = await poller.start();
+
+	await unsubscribeToUpdateMessages(
+		simulationIds,
+		ClientEventType.SimulationSciml,
+		getMessageHandler
+	);
+
+	if (pollerResults.state !== PollerState.Done || !pollerResults.data) {
+		// throw if there are any failed runs for now
+		showSpinner.value = false;
+		logger.error(`Calibrate: ${simulationIds} has failed`, {
+			toastTitle: 'Error - Julia'
+		});
+		throw Error('Failed Runs');
+	}
+	completedRunIdList.value = simulationIds;
+
+	runInProgress.value = undefined;
+	showSpinner.value = false;
+};
+
+const watchCompletedRunList = async (runIdList: string[]) => {
+	if (runIdList.length === 0) return;
+
+	const newRunResults = {};
+	const newRunResultParams = {};
+	await Promise.all(
+		runIdList.map(async (runId) => {
+			if (runResults.value[runId] && runResultParams.value[runId]) {
+				newRunResults[runId] = runResults.value[runId];
+				newRunResultParams[runId] = runResultParams.value[runId];
+			} else {
+				const result = await getRunResultJulia(runId, 'result.json');
+				if (result) {
+					const csvData = csvParse(result.csvData);
+					newRunResults[runId] = csvData;
+					newRunResultParams[runId] = result.paramVals;
+				}
+			}
+		})
+	);
+	runResults.value = newRunResults;
+	runResultParams.value = newRunResultParams;
+
+	const port = props.node.inputs[0];
+
+	const state = _.cloneDeep(props.node.state);
+
+	state.calibrateConfigs.runConfigs[runIdList[0]] = {
+		runId: runIdList[0],
+		active: true,
+		loss: lossValues
+	};
+	emit('update-state', state);
+
+	// clear out intermediate values for next run
+	lossValues = [];
+	parameterResult.value = {};
+
+	emit('append-output-port', {
+		type: CalibrationOperationJulia.outputs[0].type,
+		label: `${port.label} - Output ${runList.value.length}`,
+		value: runIdList
+	});
+
+	// show the latest run in the dropdown
+	selectedRun.value = runList.value[runList.value.length - 1];
+};
+
+// Tom TODO: Make this generic... its copy paste from node.
```

Comment:
Outdated comment: this is no longer copy paste code as you just removed the copy

Reply:
good catch